lightly/embedding/embedding.py

```diff
@@ -94,45 +105,60 @@ def embed(self,
         embeddings, labels, fnames = None, None, []
 
         if lightly._is_prefetch_generator_available():
-            pbar = tqdm(BackgroundGenerator(dataloader, max_prefetch=3),
-                        total=len(dataloader))
+            pbar = tqdm(
+                BackgroundGenerator(dataloader, max_prefetch=3),
+                total=len(dataloader)
+            )
         else:
             pbar = tqdm(dataloader, total=len(dataloader))
 
-        efficiency = 0.
+        efficiency = 0.0
         embeddings = []
         labels = []
         with torch.no_grad():
 
-            start_time = time.time()
+            start_timepoint = time.time()
             for (img, label, fname) in pbar:
 
                 img = img.to(device)
-                label = label.to(device)
 
                 fnames += [*fname]
 
                 batch_size = img.shape[0]
-                prepare_time = time.time()
+                prepared_timepoint = time.time()
 
                 emb = self.model.backbone(img)
                 emb = emb.detach().reshape(batch_size, -1)
 
                 embeddings.append(emb)
                 labels.append(label)
 
-                process_time = time.time()
+                finished_timepoint = time.time()
+
+                data_loading_time = prepared_timepoint - start_timepoint
+                inference_time = finished_timepoint - prepared_timepoint
+                total_batch_time = data_loading_time + inference_time
 
-                efficiency = \
-                    (process_time - prepare_time) / (process_time - start_time)
-                pbar.set_description(
-                    "Compute efficiency: {:.2f}".format(efficiency))
-                start_time = time.time()
+                efficiency = inference_time / total_batch_time
+                pbar.set_description("Compute efficiency: {:.2f}".format(efficiency))
+                start_timepoint = time.time()
 
             embeddings = torch.cat(embeddings, 0)
             labels = torch.cat(labels, 0)
-            if to_numpy:
-                embeddings = embeddings.cpu().numpy()
-                labels = labels.cpu().numpy()
 
-        return embeddings, labels, fnames
+            embeddings = embeddings.cpu().numpy()
+            labels = labels.cpu().numpy()
+
+        to_order = zip(embeddings, labels)
```

Comment:
Sorry I didn't mention that before:
- I think you are trying to reduce redundant code (which is great) by using zip (which is a little hard to read)
- I think a cleaner way would be to use a function which orders any list by any other list:

```python
def sort_items_by_keys(
    keys: List[any],
    items: List[any],
    sorted_keys: List[any]
):
    lookup = { key_: item_ for key_, item_ in zip(keys, items) }
    sorted_ = [lookup[key_] for key_ in sorted_keys]
    return sorted_

Reply:
And then 
```python
sorted_filenames = dataloader.dataset.get_filenames()
sorted_embeddings = sort_items_by_keys(fnames, embeddings, sorted_filenames)
sorted_labels = sort_items_by_keys(fnames, labels, sorted_filenames)
return ....
```